org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/points/GeneratorFilter.java

```diff
@@ -78,12 +79,14 @@ public GeneratorFilter(IPointGeneratorService pservice, SeriesTable table, IAdap
 	}
 
 	
-	public List<Object> getModels(List<ISeriesItemDescriptor> seriesItems) throws Exception {
-		List<Object> models = new ArrayList<>();
+	@SuppressWarnings("unchecked")
+	public <T> List<T> getModels(List<ISeriesItemDescriptor> seriesItems) {
+		if (seriesItems==null) return null;

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)